vignettes/convert.qmd

```diff
@@ -15,13 +15,70 @@ execute:
+Learning to use `DuckDB` and `Parquet` is easy for anyone who have ever worked with `{dplyr}` functions such as `select()`, `filter()`, `mutate()`, `group_by()`, `summarise()`, etc. However, since there is some learning curve to master these new tools, we provide some helper functions for novices to get started and easily open the datasets from `DuckDB` and `Parquet`. Please read the relevant sections below, where we first show how to convert the data, and then how to use it.
+
+## How to choose between DuckDB, Parquet, and CSV {#duckdb-vs-parquet-csv}
+
+### Speed {#speed-comparison}
+
+In our testing (see @fig-csv-duckdb-parquet-speed), using `DuckDB` database has a significant speed advantage over running the analysis on raw CSV.gz files. Using the full origin destination dataset for 18 months, we compared a query that would find mean hourly number of trips between all zone pairs over the whole time period. `DuckDB` database was up to 8 times faster than running the analysis on raw CSV.gz files, and up to 5 times faster than running the analysis on `Parquet` files.
+
+![Data processing speed comparison: DuckDB engine running on CSV.gz files vs DuckDB database vs a folder of Parquet files](media/duckdb-parquet-csv-speed-mean-hourly-v1.svg){#fig-csv-duckdb-parquet-speed}
+
+
+<details><summary>You can see the query we used for measuring the speed here</summary>
+
+For reference, here is a simple query we used for speed comparison in @fig-csv-duckdb-parquet-speed:
+
+```r
+# data represents either CSV files acquired from `spod_get()`, a `DuckDB` database or a folder of Parquet files connceted with `spod_connect()`
+data |>
+  group_by(id_origin, id_destination, time_slot) |> 
+  summarise(mean_hourly_trips = mean(n_trips, na.rm = TRUE),
+    .groups = "drop")
+```
+
+</details>
+
+
+@fig-csv-duckdb-parquet-speed also shows that `DuckDB` format will give you the best performance even on low-end systems with limited memory and number of processor cores, conditional on a fast SSD storage. Also note, that if you do choose to work with long time periods using CSV.gz files via `spod_get()`, you will need to balance the amount of memory and processor cores via the `max_n_cpu` and `max_mem_gb` arguments, otherwise the analysis may fail (see the grey area in the figure), when there are too many parallel processes running at the same time with limited memory.
+
+### Convenience
+
+The approaches to working with all formats are exactly the same, as under the hood all analysis is actually done by the `DuckDB` [@duckdb-r] engine. So from that point of view, there is no difference between data formats. You can manipulate the data using `{dplyr}` functions such as `select()`, `filter()`, `mutate()`, `group_by()`, `summarise()`, etc. In the end of any sequence of commands you will need to add `collect()` to execute the whole chain of data manipulations and load the results into memory in an R `data.frame`/`tibble`. We provide examples in the following sections.
```

Comment:
@eugenividal what kind of explanation...? We already have a link/reference to what duckdb is. And we explain in the next paragraph how to work with it, which I assumed should be just enough information at this point. What would you like to see here?

Reply:
@e-kotov When you say all approaches you mean DuckDB, Parquet, and CSV?

Reply:
@eugenividal yes

Reply:
@eugenividal perhaps the vague "approaches" should be changed literally to functions and dplyr

Reply:
@eugenividal would that be better? *"Independant of the data format (`DuckDB`, `Parquet`, or `CSV.gz`). The functions you will need to analyse the data are exactly the same, as under the hood all analysis is actually done by the `DuckDB` [@duckdb-r] engine that presents the data as if were a regular `data.frame`/`tibble` object (almost)."*

Reply:
@e-kotov Yes! That sounds clearer. I made some very small changes to your sentence (use only if you like it): "Regardless of the data format (DuckDB, Parquet, or CSV.gz), the functions you will need for data manipulation and analysis are the same. This is because the analysis is actually performed by the DuckDB engine, which presents the data as if it were a regular data.frame/tibble object (almost)." Spaniards also like long sentences, but I am afraid British do not.

Reply:
@eugenividal Love that